### ISSUE TITLE
Rename apk when flutter build 

### ISSUE THREAD

**Issue by nodahikaru** [flutter/flutter#29509]
Is there any way to change apk name instead of default app-debug.apk when run "flutter build apk --debug or --release"?

**Comment by areeb111** [comment]
http://android-ed.blogspot.com/2015/07/change-apk-file-name-in-gradle.html
Check this out

**Comment by nodahikaru** [comment]
@areeb111 I already used it. But when I use it in android project inside Flutter project, I got this error 
"Gradle build failed to produce on Android package", but apk is built successfully. 
It's very important because it runs on bitbucket pipeline.

**Comment by raphaelsoul** [comment]
> @areeb111 I already used it. But when I use it in android project inside Flutter project, I got this error
> "Gradle build failed to produce on Android package", but apk is built successfully.
> It's very important because it runs on bitbucket pipeline.

archivesBaseName changes but flutter still trying to find it on `app-debug.apk` or `app-release.apk`. Yet havn't find solution

**Comment by patelkrunal78** [comment]
The Flutter documentation points out where you can change the display name of your application for both Android and iOS. This may be what you are looking for:

Preparing an Android App for Release
Preparing an iOS App for Release
For Android

It seems you have already found this in the AndroidManifest.xml as the application entry.

Review the default App Manifest file AndroidManifest.xml located in /android/app/src/main/ and verify the values are correct, especially:

application: Edit the application tag to reflect the final name of the app.

Find More [Here](https://icetutor.com/question/how-to-change-the-app-display-name-build-with-flutter)

**Comment by hundeva** [comment]
For the record, this is still an issue, not being able to change the apk name adds an extra manual step on builds, if you want to keep a history of apks for example.

**Comment by raphaelsoul** [comment]
> For the record, this is still an issue, not being able to change the apk name adds an extra manual step on builds, if you want to keep a history of apks for example.

yes that is exactly what I want

**Comment by mkulesh** [comment]
For the record, I have the same issue. In gradle configuration, I defined
`buildTypes {
        release {
            signingConfig signingConfigs.release
            setProperty("archivesBaseName", "onpc-v" + flutterVersionName)
        }
    }
`

and the APK  onpc-v2.0.0-release.apk is successfully created in the /build/app/outputs/apk/release directory, but Flutter generates an error "Finished with error: Gradle build failed to produce an Android package" and is not able to automatically install package on device.

**Comment by amorenew** [comment]
I don't know that we can change apk name.
What about adding version number the apk?
like Facebook V0.0.14 Release.APK

**Comment by blasten** [comment]
I don't recommend changing the name since the Flutter tool won't be able to find this APK file. It will still generate the APK, but won't print success.

**Comment by jonahgreenthal** [comment]
@blasten Presumably the point of this issue is that it should be possible to automatically rename the file _without_ borking Flutter's build process.

**Comment by amreniouinnovent** [comment]
@blasten Actually same issue in iOS there is an issue related to "Runner Target", Developers need flexibility and I think as Flutter supports more platform so more and more automation, a smart building is required.

**Comment by jmagman** [comment]
iOS renames are being tracked in https://github.com/flutter/flutter/issues/9767 and https://github.com/flutter/flutter/issues/14648.  

This issue is specifically regarding APKs.

**Comment by skordesign** [comment]
I got same issue, any update?

**Comment by chitgoks** [comment]
still persists. code that worked in native android project will generate that same error message when used in flutter project.

**Comment by jmagman** [comment]
@chitgoks If you see an error message you should file a new issue and completely fill out the template.  This issue is tracking a rename request, not an error.

**Comment by chitgoks** [comment]
> @chitgoks If you see an error message you should file a new issue and completely fill out the template. This issue is tracking a rename request, not an error.

you are right. this was supposed to be a reply from a comment above the thread.

**Comment by veneno261** [comment]
I have flavor in my project and after changing APK file name and run debug application this error appears in the console:

> Exception: Gradle build failed to produce an .apk file. It's likely that this file was generated under C:\projects\my_app_directory\build, but the tool couldn't find it.

I found this error throw because flutter only can find a specific APK name when we run the app in debug mode
only can find this name `app-FLAVORNAME-debug.apk`

and with this code in Gradle, I separate debug and release APK name

````
buildTypes {
        release {
            signingConfig signingConfigs.debug
            applicationVariants.all { variant ->
                variant.outputs.all {
                    def appName = "your_app_name_"
                    def buildType = variant.variantData.variantConfiguration.buildType.name
                    def newName
                    if (buildType == 'debug'){
                        newName = "app-${variant.getFlavorName()}-debug.apk"
                    } else {
                        newName = "${appName}${defaultConfig.versionName}_${variant.getFlavorName()}.apk"
                    }
                    outputFileName = newName
                }
            }
        }
    }
````

and after run debug the file name should be `app-flavor-debug.apk` and release filename should be  `your_app_name_version_flavor.apk`

**Comment by XuZhaoTong** [comment]
@veneno261 
I tried,if the name of the apk is not 'app-flavor-realse.apk', it will report an error of "Gradle build failed to produce an .apk file. It's likely that this file was generated under /build, but the tool couldn't find it."

**Comment by raphaelsoul** [comment]
One year past.
The flavor method seems to be more like a workaround rather than a tool chain solution. And yet have no solution to resolve it by locating in flutter tool chain.

I recommend to reopen this issue. @nodahikaru @blasten @jonahwilliams

**Comment by NO-ob** [comment]
Bumping, this would be a nice feature

**Comment by dheerajmpai** [comment]
It would be really nice if this can be passed in the CLI

**Comment by Marty** [comment]
This is important for our automated deployment and archiving via gitlab-ci.

**Comment by moseskarunia** [comment]
Hi guys, I write some script utilizing `yq` to rename my apk after build. It's not elegant, but at least it does the job.

If you want to run it, run it post build in your CI pipeline. I also successfully configured my codemagic pipeline with this.

```sh
mv $RELEASE_BUNDLE_PATH/app-release.aab $RELEASE_BUNDLE_PATH/$(cat pubspec.yaml | yq e '.name' -)_$(cat pubspec.yaml | yq e '.version | split("+")[1] + "_" + split("+")[0]' -)-release.aab
```

Btw in codemagic, you need to run this script first.

```sh
#!/usr/bin/env sh
HOMEBREW_NO_AUTO_UPDATE=1 brew update && brew upgrade yq
```

This is because the yq's split function only available in 4.4.0 and the one in codemagic is not yet 4.4.0. The brew update & upgrade should take around 1m normally.
